File: MATLAB/function_make_script_array.m
function jobind=function_make_script_array(scriptToRun,noruns,stind)

%Make output directories if needed
mkdir jobs
mkdir logs

jobind=stind;
display('Making jobs ...')

jobind=jobind+1;
fid = fopen(['./jobs/job_' num2str(jobind) '.sh'],'w');

fprintf(fid,'#!/bin/bash\n\n');

fprintf(fid,'#SBATCH -p batch\n');
fprintf(fid,'#SBATCH -t 1:59:59\n'); %  Time limits
fprintf(fid,['#SBATCH --array=1-' num2str(noruns) '\n\n']); % How many repetitions in the array?
fprintf(fid,['#SBATCH -o ' './logs/log_' num2str(jobind) '_%%a\n']);
fprintf(fid,'#SBATCH --qos=normal\n\n');
fprintf(fid,'#SBATCH --mem-per-cpu=10000\n\n');

fprintf(fid,'sleep .$[( $RANDOM %% 10 ) ]s\n\n'); % Sleep for a random amount of time, so that the rng will give different seed
fprintf(fid,['matlab -nojvm -r "cd ' pwd '/;rng(''shuffle'');' scriptToRun ';exit;"']); % for 32mm

fclose(fid);
display('Done making jobs!')
end

File: Python/function_make_script_array.py
import os
import argparse

os.makedirs('jobs',exist_ok = True)
os.makedirs('logs',exist_ok = True)

parser = argparse.ArgumentParser(description='Process some integers.')
# Add an argument
parser.add_argument('--script', type=str, default='echo "No command provided!"')
parser.add_argument('--jobind', type=int, default=0)
parser.add_argument('--noruns', type=int, default=10)

args = parser.parse_args()

script = args.script
jobind = args.jobind;
noruns = args.noruns

curpath = os.getcwd()

os.system('rm ' + curpath + ' /logs/*')
os.system('rm ' + curpath + '/jobs/*')

def function_make_script_array(script='echo "Test Run"',noruns=10,jobind=0):
    print('Making jobs...')
    curdir = os.getcwd()
    jobind=jobind+1
    f = open('jobs/job_' + str(jobind) + '.sh', 'w')
    f.write("#!/bin/bash\n\n")
    f.write("#SBATCH -t 1:59:59\n")
    f.write('#SBATCH --array=1-' + str(noruns) + '\n\n');
    f.write('#SBATCH -o ./logs/log_' + str(jobind) + '\n')
    f.write("#SBATCH --qos=normal\n\n")
    f.write("#SBATCH --mem-per-cpu=31000\n\n")
    f.write('echo' + script + '; cd '+ curpath + '; ' + script)
    f.close()
    return jobind

if __name__ == '__main__':
    make_jobs('echo "Testing SLURMMM..."', noruns=noruns, jobind=jobind)